h2o-core/src/main/java/water/fvec/Vec.java

```diff
@@ -1337,8 +1337,47 @@ public Vec align(final Vec vec) {
    *  CategoricalWrappedVec has 'this' as it's masterVec, but returns results in the 'to'
    *  domain (or just past it, if 'this' has elements not appearing in the 'to'
    *  domain). */
-  public CategoricalWrappedVec adaptTo( String[] domain ) {
-    return new CategoricalWrappedVec(group().addVec(),_rowLayout,domain,this._key);
+  public Vec adaptTo( String[] domain ) {
+    if(isInt() && ArrayUtils.isInt(domain))
+      return new CategoricalWrappedVec(group().addVec(),_rowLayout,domain,this._key);
+    final int oldDomainLen = domain.length;
+    if(isNumeric()){
+      double [] domain2 = MemoryManager.malloc8d(domain.length);
+      try {
+        for (int i = 0; i < domain2.length; ++i)
+          domain2[i] = Double.parseDouble(domain[i]);
+        Arrays.sort(domain2);
+        final double [] ddomain =new VecUtils.CollectDoubleDomain(domain2,100000).doAll(this).domain();
```

Comment:
yeah that's true. It's an adapt call so we need to preserve the original indices. Good catch.